datasource/netcdf_groundwater.py

```diff
@@ -183,157 +173,107 @@ def available_aggregation_vars(self):
-            gr = self.gridadmin_result
-            ts = self.timestamps
-        elif nc_variable in self.available_aggregation_vars:
-            gr = self.gridadmin_aggregate_result
-            ts = self.get_timestamps(parameter=nc_variable)
-        else:
-            logger.error("Unsupported variable %s", nc_variable)
+        if node_id:
+            result_filter = result_filter.filter(id=node_id)
+        elif content_pk:
+            result_filter = result_filter.filter(content_pk=content_pk)
 
-        # determine if layer is a not_schematized (e.g nodes, pumps)
-        if object_type_layer_source[object_type] == "result":
-            values = self._get_timeseries_result_layer(
-                gr, object_type, object_id, nc_variable
-            )
-        elif object_type_layer_source[object_type] == "schematized":
-            values = self._get_timeseries_schematisation_layer(
-                gr, object_type, object_id, nc_variable
-            )
+        data = result_filter.get_filtered_field_value(nc_variable)
 
-        # Zip timeseries together in (n,2) array
         if fill_value is not None:
-            # transform np.array into np.MaskedArray
-            no_data_value = -9999
-            masked_array = np.ma.masked_values(values, no_data_value)
-            filled_vals = masked_array.filled(fill_value)
-            return np.vstack((ts, filled_vals)).T
+            data[data == NO_DATA_VALUE] = fill_value
+
+        timestamps = self.get_timestamps(nc_variable)
+        timestamps = timestamps.reshape(-1, 1)
+        return np.hstack([timestamps, data])
+
+    def get_gridadmin(self, variable=None):
+        """Return the gridadmin where the variable is stored. If no variable is
+        given, a gridadmin without results is returned.
+
+        Results are either stored in the 'results_3di.nc' or the
+        'aggregate_results_3di.nc'. These make use of the GridH5ResultAdmin and
+        GridH5AggregateResultAdmin to query the data respectively.
+
+        :param variable: str of the variable name, e.g. 's1', 'q_pump'
+        :return: handle to GridAdminResult or AggregateGridAdminResult
+        """
+        if variable is None:
+            return self.gridadmin
+        elif variable in self.available_subgrid_map_vars:
+            return self.gridadmin_result
+        elif variable in self.available_aggregation_vars:
+            return self.gridadmin_aggregate_result
         else:
-            # values can contain masked values from netCDF, therefore we need
-            # np.ma.vstack
-            return np.ma.vstack((ts, values)).T
-
-    def get_timestamps(self, object_type=None, parameter=None):
-        # TODO: use cached property to limit file access
-        if parameter is None:
-            return self.ds.get("time")[:]
-        elif parameter in [v[0] for v in SUBGRID_MAP_VARIABLES]:
-            return self.ds.get("time")[:]
+            raise AttributeError("Parameter %s unknown")
```

Comment:
```suggestion
            raise AttributeError("Unknown subgrid or aggregate variable: %s")
```

- Parameter => variable.
- Iets correcter Engels.
- Explicieter over wat voor soort variabele het is.